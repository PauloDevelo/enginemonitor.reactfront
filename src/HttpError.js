export default class HttpError extends Error {
    constructor(data, ...params) {
      // Pass remaining arguments (including vendor specific ones) to parent constructor
      super(...params);
  
      // Maintains proper stack trace for where our error was thrown (only available on V8)
      if (Error.captureStackTrace) {
        Error.captureStackTrace(this, HttpError);
      }
  
      // Custom debugging information
      this.data = data;
      this.date = new Date();
    }
  }