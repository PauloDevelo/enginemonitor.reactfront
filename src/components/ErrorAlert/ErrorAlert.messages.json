{
  "actionErrorBecauseOffline": {
    "defaultMessage": "Cannot perform this action you are offline or your connection seems too slow.",
    "description": "Error message",
    "id": "actionErrorBecauseOffline"
  },
  "credentialError": {
    "defaultMessage": "You don't have enough credentials to perform this action.",
    "description": "Error message",
    "id": "credentialError"
  },
  "storageNotOpenedYet": {
    "defaultMessage": "The local storage is not opened yet.",
    "description": "Error message",
    "id": "storageNotOpenedYet"
  },
  "unexpectedError": {
    "defaultMessage": "An unexpected error occurred. You can try again.",
    "description": "Error message",
    "id": "unexpectedError"
  },
  "userExceedStorageLimit": {
    "defaultMessage": "User exceeded the storage limit. Remove some images before adding new images.",
    "description": "Error message",
    "id": "userExceedStorageLimit"
  }
}
