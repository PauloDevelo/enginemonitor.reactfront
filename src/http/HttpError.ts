// eslint-disable-next-line no-unused-vars
import { AxiosError } from 'axios';

export default class HttpError extends Error {
  private readonly date:Date;

  public readonly data:any;

  public readonly internalError:Error | undefined;

  constructor(data: any, internalError?: Error, ...params:any) {
    // Pass remaining arguments (including vendor specific ones) to parent constructor
    super(...params);

    // Maintains proper stack trace for where our error was thrown (only available on V8)
    if (Error.captureStackTrace) {
      Error.captureStackTrace(this, HttpError);
    }

    // Custom debugging information
    this.data = data;
    this.date = new Date();
    this.internalError = internalError;
  }

  public didConnectionAbort = ():boolean => {
    if (this.internalError !== undefined) {
      const { code } = this.internalError as AxiosError;
      if (code === 'ECONNABORTED') {
        return true;
      }
    }

    return false;
  };
}
