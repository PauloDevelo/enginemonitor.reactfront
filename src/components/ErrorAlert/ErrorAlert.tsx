import React, { Fragment, useState, useEffect, ReactElement } from 'react';
import { FormattedMessage } from 'react-intl';

import {Alert} from 'reactstrap';
import HttpError from '../../http/HttpError';

import {defineMessages, Messages} from "react-intl";
import jsonMessages from "./ErrorAlert.messages.json";
const alertMsg: Messages = defineMessages(jsonMessages);

type Props = {
    error: Error | undefined;
    className?: string;
    onDismiss: () => void;
}

const ErrorAlert = ({error, onDismiss, className}:Props) => {
    const [isVisible, setVisible] = useState(error !== undefined);

    useEffect(() => {
        setVisible(error !== undefined)
    }, [error]);

    let message:string = "";
    if(error !== undefined){
        if(error instanceof HttpError)
        {
            message = error.data.message;
        }
        else{
            message = error.message;
        }
    }

    let msgElement: string | ReactElement;
    if(alertMsg[message] !== undefined){
        msgElement = <FormattedMessage {...alertMsg[message]}/>;
    }
    else{
        msgElement = message;
    }

    return (
        <Fragment>
            <Alert color="danger" className={className} isOpen={isVisible} toggle={onDismiss}>
                <div className="text-center">{msgElement}</div>
            </Alert>
        </Fragment>
    )
}

export default React.memo(ErrorAlert);