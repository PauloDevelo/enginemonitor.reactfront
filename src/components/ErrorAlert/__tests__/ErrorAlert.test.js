import React from 'react';
import { mount } from 'enzyme';

import ErrorAlert from '../ErrorAlert';
import HttpError from '../../../http/HttpError';

describe('Component ErrorAlert', () => {
  it('when it gets an Error should render with a message and the default color', () => {
    // Arrange
    const error = new Error('test message');
    const onDismiss = () => {};

    // Act
    const wrapper = mount(<ErrorAlert error={error} onDismiss={onDismiss} />);

    // Assert
    expect(wrapper.text()).toEqual('×test message');
    expect(wrapper.find('div').get(0).props.className.split(' ').findIndex((i) => i === 'alert') !== -1).toEqual(true);
    expect(wrapper.find('div').get(0).props.className.split(' ').findIndex((i) => i === 'alert-danger') !== -1).toEqual(true);
    expect(wrapper).toMatchSnapshot();
  });

  it('when it gets an HttpError should render with a message and the default color', () => {
    // Arrange
    const error = new HttpError({ message: 'test message' });
    const onDismiss = () => {};

    // Act
    const wrapper = mount(<ErrorAlert error={error} onDismiss={onDismiss} />);

    // Assert
    expect(wrapper.text()).toEqual('×test message');
    expect(wrapper.find('div').get(0).props.className.split(' ').findIndex((i) => i === 'alert') !== -1).toEqual(true);
    expect(wrapper.find('div').get(0).props.className.split(' ').findIndex((i) => i === 'alert-danger') !== -1).toEqual(true);
    expect(wrapper).toMatchSnapshot();
  });
});
