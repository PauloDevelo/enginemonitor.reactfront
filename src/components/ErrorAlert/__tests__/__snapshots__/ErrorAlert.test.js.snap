// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component ErrorAlert when it gets an Error should render with a message and the default color 1`] = `
<Memo(ErrorAlert)
  error={[Error: test message]}
  onDismiss={[Function]}
>
  <Alert
    closeAriaLabel="Close"
    color="danger"
    fade={true}
    isOpen={true}
    tag="div"
    toggle={[Function]}
    transition={
      Object {
        "appear": true,
        "baseClass": "fade",
        "baseClassActive": "show",
        "enter": true,
        "exit": true,
        "in": true,
        "mountOnEnter": false,
        "onEnter": [Function],
        "onEntered": [Function],
        "onEntering": [Function],
        "onExit": [Function],
        "onExited": [Function],
        "onExiting": [Function],
        "tag": "div",
        "timeout": 150,
        "unmountOnExit": true,
      }
    }
  >
    <Fade
      appear={true}
      baseClass="fade"
      baseClassActive="show"
      className="alert alert-danger alert-dismissible"
      enter={true}
      exit={true}
      in={true}
      mountOnEnter={false}
      onEnter={[Function]}
      onEntered={[Function]}
      onEntering={[Function]}
      onExit={[Function]}
      onExited={[Function]}
      onExiting={[Function]}
      role="alert"
      tag="div"
      timeout={150}
      unmountOnExit={true}
    >
      <Transition
        appear={true}
        enter={true}
        exit={true}
        in={true}
        mountOnEnter={false}
        onEnter={[Function]}
        onEntered={[Function]}
        onEntering={[Function]}
        onExit={[Function]}
        onExited={[Function]}
        onExiting={[Function]}
        timeout={150}
        unmountOnExit={true}
      >
        <div
          className="alert alert-danger alert-dismissible fade"
          role="alert"
        >
          <button
            aria-label="Close"
            className="close"
            onClick={[Function]}
            type="button"
          >
            <span
              aria-hidden="true"
            >
              ×
            </span>
          </button>
          <div
            className="text-center"
          >
            test message
          </div>
        </div>
      </Transition>
    </Fade>
  </Alert>
</Memo(ErrorAlert)>
`;

exports[`Component ErrorAlert when it gets an HttpError should render with a message and the default color 1`] = `
<Memo(ErrorAlert)
  error={[Error]}
  onDismiss={[Function]}
>
  <Alert
    closeAriaLabel="Close"
    color="danger"
    fade={true}
    isOpen={true}
    tag="div"
    toggle={[Function]}
    transition={
      Object {
        "appear": true,
        "baseClass": "fade",
        "baseClassActive": "show",
        "enter": true,
        "exit": true,
        "in": true,
        "mountOnEnter": false,
        "onEnter": [Function],
        "onEntered": [Function],
        "onEntering": [Function],
        "onExit": [Function],
        "onExited": [Function],
        "onExiting": [Function],
        "tag": "div",
        "timeout": 150,
        "unmountOnExit": true,
      }
    }
  >
    <Fade
      appear={true}
      baseClass="fade"
      baseClassActive="show"
      className="alert alert-danger alert-dismissible"
      enter={true}
      exit={true}
      in={true}
      mountOnEnter={false}
      onEnter={[Function]}
      onEntered={[Function]}
      onEntering={[Function]}
      onExit={[Function]}
      onExited={[Function]}
      onExiting={[Function]}
      role="alert"
      tag="div"
      timeout={150}
      unmountOnExit={true}
    >
      <Transition
        appear={true}
        enter={true}
        exit={true}
        in={true}
        mountOnEnter={false}
        onEnter={[Function]}
        onEntered={[Function]}
        onEntering={[Function]}
        onExit={[Function]}
        onExited={[Function]}
        onExiting={[Function]}
        timeout={150}
        unmountOnExit={true}
      >
        <div
          className="alert alert-danger alert-dismissible fade"
          role="alert"
        >
          <button
            aria-label="Close"
            className="close"
            onClick={[Function]}
            type="button"
          >
            <span
              aria-hidden="true"
            >
              ×
            </span>
          </button>
          <div
            className="text-center"
          >
            test message
          </div>
        </div>
      </Transition>
    </Fade>
  </Alert>
</Memo(ErrorAlert)>
`;
